roles/kubernetes-apps/ansible/tasks/coredns.yml

```diff
@@ -0,0 +1,34 @@
+---
+- name: Kubernetes Apps | Lay Down CoreDNS Template
+  template:
+    src: "{{ item.file }}.j2"
+    dest: "{{ kube_config_dir }}/{{ item.file }}"
+  with_items:
+    - { name: coredns, file: coredns-config.yml, type: configmap }
+    - { name: coredns, file: coredns-sa.yml, type: sa }
+    - { name: coredns, file: coredns-deployment.yml, type: deployment }
+    - { name: coredns, file: coredns-svc.yml, type: svc }
+    - { name: coredns, file: coredns-clusterrole.yml, type: clusterrole }
+    - { name: coredns, file: coredns-clusterrolebinding.yml, type: clusterrolebinding }
+  register: coredns_manifests
+  when:
+    - dns_mode in ['coredns', 'coredns_dual']
+    - inventory_hostname == groups['kube-master'][0]
+    - rbac_enabled or item.type not in rbac_resources
+  tags:
+    - coredns
+
+- name: Kubernetes Apps | Lay Down Secondary CoreDNS Template
```

Comment:
On this task, you could set a task only var `coredns_secondary_tmpl: true` or something, and target the same same template file as the above task. Something like
```yaml
- name: Kubernetes Apps | Lay Down Secondary CoreDNS Template
  template:
    src: "{{ item.file }}.j2"
    dest: "{{ kube_config_dir }}/{{ item.file }}"
  with_items:
    - { name: coredns, file: coredns-deployment.yml, type: deployment }
    - { name: coredns, file: coredns-svc.yml, type: svc }
  vars:
    coredns_ordinal: "-secondary"
...
```

Reply:
Wont be possible to do so. It would overwrite the file with the first svc and deployment in it?

Reply:
```yaml
- name: Kubernetes Apps | Lay Down Secondary CoreDNS Template
  template:
    src: "{{ item.src}}"
    dest: "{{ kube_config_dir }}/{{ item.file }}"
  with_items:
    - { name: coredns, file: coredns-deployment-secondary.yml, src: coredns-deployment.yml.j2, type: deployment }
    - { name: coredns, file: coredns-svc-secondary.yml, src: coredns-svc.yml.j2 , type: svc }
  vars:
    coredns_ordinal: "-secondary"
...
```